subprojects/resources/src/main/java/org/gradle/internal/resource/EmptyFileTextResource.java

```diff
@@ -20,15 +20,20 @@
 import org.gradle.internal.hash.HashCode;
 import org.gradle.internal.hash.Hashing;
 
+import javax.annotation.Nonnull;
```

Comment:
No, though there are a bunch of classes there already, don't know if I can make that promise after the fact

Reply:
It's all internal API so it should be doable without too much repercussion. But it's definitely not critical.

Reply:
The repercussion I was concerned with was leading the reader to believe nulls are not expected anywhere (unless otherwise stated), when I cannot tell for sure myself whether the intent for every single parameter in that package is for it not to be nullable.